face_landmark/webgpu/js/app.js

```diff
@@ -0,0 +1,870 @@
+        }
+    });
+}
+
+// Helper function to wait for video to be ready
+function waitForVideoReady(video) {
+    return new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => {
+            reject(new Error('Video ready timeout'));
+        }, 10000);
+
+        function checkVideo() {
+            if (isVideoReady(video)) {
+                clearTimeout(timeout);
+                resolve();
+            } else {
+                requestAnimationFrame(checkVideo);
+            }
+        }
+        checkVideo();
+    });
+}
+
+
+async function init() {
+    try {
+        // 1. Check WebGPU support
+        if (!navigator.gpu) {
+            throw new Error('WebGPU not supported');
+        }
+
+        // 2. Check dependencies
+        try {
+            checkDependencies();
+        } catch (error) {
+            console.error('Dependency check failed:', error);
+            document.getElementById('debug_log').innerHTML = `Error: ${error.message}`;
+            return;
+        }
+
+        // 3. Initialize WebGPU renderer FIRST
+        const canvas = document.getElementById('gpuCanvas');
+        renderer = new FaceLandmarkRenderer();
+        await renderer.initialize(canvas);
+        console.log('WebGPU renderer initialized');
+
+        // 4. Initialize TensorFlow.js
+        await tf.ready();
+        console.log('TensorFlow.js ready');
+
+        // 5. Initialize face detection
+        if (typeof faceLandmarksDetection === 'undefined') {
+            throw new Error('Face landmarks detection library not loaded');
+        }
+
+        try {
+            faceMeshModel = await faceLandmarksDetection.createDetector(
+                faceLandmarksDetection.SupportedModels.MediaPipeFaceMesh,
+                {
+                    runtime: 'tfjs',
+                    refineLandmarks: true,
+                    maxFaces: 1,
+                }
+            );
+            console.log('Face mesh model loaded');
+        } catch (modelError) {
+            console.error('Error loading face mesh model:', modelError);
+            throw modelError;
+        }
+
+        // 6. Initialize video stream with proper setup
+        console.log('Setting up video stream...');
+        try {
+            videoStream = await setupVideoStream();
+            console.log('Video stream initialized');
+
+            await waitForVideoReady(videoStream);
+            console.log('Video stream ready', {
+                width: videoStream.videoWidth,
+                height: videoStream.videoHeight
+            });
+
+        } catch (streamError) {
+            console.error('Error accessing camera:', streamError);
+            throw streamError;
+        }
+
+        // 7. Initialize texture loader and mask manager
+        // Now renderer.device is available
+        const textureLoader = await TextureLoader.initialize(renderer.device);
+            
+        const maskManager = new MaskManager(renderer.device, textureLoader);
```

Comment:
Since ```maskManger``` is a local value, it becomes invalid when this function exits.
it may work if you remove ```const``` keyword.